File: C/THE_ULTIMATE_AVG_CALC.c
#include <stdio.h>
int main(){
    int coofecients[]={6,5,5,3,3,3,1,1,1,1,1}; 
    char *modules[]={"analysis","physics","chemistry","algebra","computer_science","statistics","technical_drawing","french","english","economics","human_engineering"};
    float avg_s1=0,avg_s2=0,AVG,score[11],exam_marks[11],control_marks[11];
 printf("***welcome to the average calculator for 1st year preparatory school here you will know if you are good or cooked***\n\n\n");

           printf("*welcome to the first semester*\n\n");
           printf("your modules are:\n");
           for(i=0;i<11;i++){
    printf("%d - %s with coefficient %d\n",i+1,modules[i],coofecients[i]);}
    

    printf("let's begin with exam marks\n\n");
    printf("give your exam mark for module ");
    for(i=0;i<11;i++){
        printf("%s=",modules[i]);
        scanf("%f",&exam_marks[i]);
        while(exam_marks[i]<0 || exam_marks[i]>20){
            printf("invalid mark please give a mark between 0 and 20\n\n");
            printf("give your exam mark for module %s=",modules[i]);
            scanf("%f",&exam_marks[i]);
        }
    }
    
    
    printf("let's now go for control marks\n\n");
    printf("give your control mark for module ");
    for(i=0;i<11;i++){
        printf("%s=",modules[i]);
        scanf("%f",&control_marks[i]);
        while(control_marks[i]<0 || control_marks[i]>20){
            printf("invalid mark please give a mark between 0 and 20\n\n");
            printf("give your control mark for module %s=",modules[i]);
            scanf("%f",&control_marks[i]);
        }}
    
       for(i=0;i<11;i++){
           score[i]=(exam_marks[i]*0.6+control_marks[i]*0.4)*coofecients[i];
       }
    printf("let's now calculate your average for the first semester\n\n");
    for(i=0;i<11;i++){
        avg_s1=avg_s1+score[i];
    }
    avg_s1=avg_s1/(6+5+5+3+3+3+1+1+1+1+1);
    printf("your average for the first semester is %f\n\n",avg_s1);
   
    

    
    
    for(i=0;i<11;i++){
        score[i]=0;
    }


        printf("welcome to the second semester\n\n");

       modules[5]="probability";
       coofecients[0]=5;
       coofecients[3]=4;

printf("your modules are:\n");
for(i=0;i<11;i++){
    printf("%d - %s with coefficient %d\n",i+1,modules[i],coofecients[i]);}


printf("let's begin with exam marks\n\n");
    printf("give your exam mark for module ");
   for(i=0;i<11;i++){
        printf("%s=",modules[i]);
        scanf("%f",&exam_marks[i]);
        while(exam_marks[i]<0 || exam_marks[i]>20){
            printf("invalid mark please give a mark between 0 and 20\n");
            printf("give your exam mark for module %s=",modules[i]);
            scanf("%f",&exam_marks[i]);
        }}

 printf("let's now go for control marks\n\n");
    printf("give your control mark for module ");
 for(i=0;i<11;i++){
        printf("%s=",modules[i]);
        scanf("%f",&control_marks[i]);
    while(control_marks[i]<0 || control_marks[i]>20){
            printf("invalid mark please give a mark between 0 and 20\n");
            printf("give your control mark for module %s=",modules[i]);
            scanf("%f",&control_marks[i]);
        }
    }
    for(i=0;i<11;i++){
           score[i]=(exam_marks[i]*0.6+control_marks[i]*0.4)*coofecients[i];
}
printf("let's now calculate your average for the second semester\n");
    for(i=0;i<11;i++){
        avg_s2=avg_s2+score[i];}
        avg_s2=avg_s2/(5+5+5+4+3+3+1+1+1+1+1);
        printf("your average for the second semester is %f\n\n\n\n",avg_s2);
      
        
        AVG=(avg_s1+avg_s2)/2;
        printf("*your average for the year is.............  %f !*\n\n",AVG);
        if(AVG>=16){
            printf("amazing you are such a nerd");
        }
        else{
            if(AVG<10){
                printf("idiot you will have to repeat the year haha");
            }
            else{printf("good job!");}
        }
    



    return 0;}

File: Python/THE_ULTIMATE_AVG_CALC.py
print("\n\n****Welcome to the Ultimate Average Calculator! for the first year prepa cycle****\n\n")
coefecient =[6,5,5,3,3,3,1,1,1,1,1]
exam_marks=[0]*11
control_marks=[0]*11
score=[0]*11
modules = ["Analysis","Physics","Chemistry","Statistics","Algebra","Computer_Scince","Technical_Drawing","English","French","Economy","Human_Enginnering"]
avg_s1=0
avg_s2=0
avg=0
print("**Let's start with the first semester**\n")
print("your modules are:\n")
for i in range(11):
       print(f"{modules[i]} with coefecient {coefecient[i]}\n ")
print("give your exam marks\n")
for i in range(11):
      exam_marks[i]=float(input(f"{modules[i]}="))
      while exam_marks[i]<0 or exam_marks[i]>20 :
             print("\n!*invalid marks please give a mark between 0 and 20*!\n")
             exam_marks[i]=float(input(f"{modules[i]}="))
print("now give your control marks\n")
for i in range(11):
       control_marks[i]=float(input(f"{modules[i]}="))    
       while control_marks[i]<0 or control_marks[i]>20 :
              print("\n!*invalide marks plesae give a marks between 0 and 20*!\n")
for i in range(11):
       score[i]=(exam_marks[i]*0.6+control_marks[i]*0.4)*coefecient[i]
       avg_s1=avg_s1+score[i]
avg_s1=avg_s1/(30)       
print(f"\n\nyour average for the first semester is ...... {avg_s1}\n\n")
print("**Now we go for the second semester**\n")
modules[3]="Probability"
coefecient[0], coefecient[4]=5, 4
print("your modules are:\n")
for i in range(11):
       print(f"{modules[i]} with coefecient {coefecient[i]}\n ")
print("give your exam marks\n")
for i in range(11):
      exam_marks[i]=float(input(f"{modules[i]}="))
      while exam_marks[i]<0 or exam_marks[i]>20 :
             print("\n!*invalid marks please give a mark between 0 and 20*!\n")
             exam_marks[i]=float(input(f"{modules[i]}="))
print("now give your control marks\n")
for i in range(11):
       control_marks[i]=float(input(f"{modules[i]}="))    
       while control_marks[i]<0 or control_marks[i]>20 :
              print("\n!*invalide marks plesae give a marks between 0 and 20*!\n")
for i in range(11):
       score[i]=(exam_marks[i]*0.6+control_marks[i]*0.4)*coefecient[i]
       avg_s2=avg_s2+score[i]
avg_s2=avg_s2/(30)       
print(f"\n\nyour average for the second semester is ...... {avg_s2}\n\n")
avg=(avg_s1+avg_s2)/2
print(f"**your average for the first year prepa cycle is**........................ !!! {avg} !!!")
if avg>17.5 :
       print("*!! THE WINNER TAKES IT ALL !!*")
elif avg>16 :
       print("good job you are such a nerd!")
elif avg>10 :
       print("you have to work more idiot")
else :
       print("LOSSER\n you have to wait another year to pass NIGGA")             
